src/hackathons-collector/hackathons-collector.js

```diff
@@ -24,31 +25,48 @@ import akai from "./assets/akai-logo.svg";
 export default class HackathonsCollector extends Component {
   constructor() {
     super();
-    this.previousEvents = events
+    this.state = {
+      filterWeekends: false,
+      allEvents: events
+    };
+  }
+  render() {
+    const previousEvents = this.state.allEvents
       .filter(event => event.date <= new Date())
+      .filter(event => {
+        return !this.state.filterWeekends || event.date.getDay() > 5;
+      })
       .sort((a, b) => a.date - b.date)
       .reverse()
       .splice(0, 3);
-
-    this.events = events
+    const events = this.state.allEvents
       .filter(event => event.date > new Date())
+      .filter(event => {
+        return !this.state.filterWeekends || event.date.getDay() > 5;
+      })
       .sort((a, b) => a.date - b.date);
-  }
-  render() {
     return (
       <div>
         <Header>
           <Logo src={logo} />
           <H1>Hackathons Collector</H1>
         </Header>
+        <Filter
+          value={this.state.filterWeekends}
+          setValue={newValue => {
+            this.setState({ filterWeekends: newValue });
+          }}
+        >
+          Only weekend events
```

Comment:
Filter component does not take children prop 😉We could pass that and make it quite reusable